app/models/linkset.rb

```diff
@@ -13,6 +13,14 @@ def empty?
   end
 
   def update_attributes_from_gem_specification!(spec)
-    self.update_attributes!(:home => spec.homepage)
+    meta = spec.metadata || {}
```

Comment:
Final answer: If there is no `metadata` in the `Gemfile`, `spec.metadata == nil`. This safeguards it.